synapse/app/_base.py

```diff
@@ -151,7 +170,7 @@ def run():
         run()
 
 
-def quit_with_error(error_string: str) -> NoReturn:
+def quit_with_error(error_string: str) -> None:
```

Comment:
I think `NoReturn` was correct, because we call `sys.exit()`.

Reply:
Doesn't `NoReturn` imply an exception is raised? Although maybe that's how `sys.exist()` works...

Reply:
The docs just say "it won't return", which is a bit ambiguous. I think it's meant to enable detection of unreachable code?

My understanding is that if `f: Callable[..., NoReturn]` then in the following sample

- `x` will never be assigned a value
- `foo()` will never be called

```python
x = f()
foo()
```

FWIW typeshed's [stub](https://github.com/python/typeshed/blob/ab25042efa1c325d242147c51d3cf28478c5a36d/stdlib/sys.pyi#L171) for sys.exit uses NoReturn; see also https://github.com/python/mypy/issues/2280.